datashader/glyphs/line.py

```diff
@@ -1437,10 +1437,14 @@ def extend_cpu_numba(
                 segment_start = (
                         (j == 0) or
                         isnull(x_flat[x_start_index + j - 1]) or
-                        isnull(y_flat[y_start_index + j] - 1)
```

Comment:
Note that the location of the closing square bracket has been wrong here for quite a long time.

Reply:
Yikes! With what effect?

Reply:
Fortunately the incorrect array index was always in bounds, so wouldn't have caused a segmentation violation! 

The `isnull()` is essentially a `nan` check, so subtracting the `1` makes no difference. Hence it was looking one index out to identify if a line segment between 2 (x, y) points is the start of a new line or not, and could have made the wrong decision. Before antialiased lines the `segment_start` flag was only used to decide whether to render the start pixel, so this error could have double-rendered some pixels.

You could probably design a pathological case to cause a line to be completely double rendered. But it would only ever have affected those using `nan` coords in ragged arrays, and if you are using ragged arrays you don't ever need to use `nan`s anyway. So it probably didn't affect any real user such that they would notice.